### PR TITLE
Rolling sha256 jets

### PR DESCRIPTION
**Pull request by roconnor-blockstream**
Adds jets to manipulate a Sha-256 context.

### CODE REVIEW

**Comment by roconnor-blockstream** [comment]
@apoelstra requesting a review of the C code.

File: C/jets.c [hunk 1]
```diff
@@ -80,6 +91,165 @@ bool sha_256_block(frameItem* dst, frameItem src, const txEnv* env) {
   return true;
 }
 
+static void read_buffer8(unsigned char* buf, size_t* len, frameItem* src, int n) {
```
Comment (roconnor-blockstream) [inline C/jets.c:94]: Add some comments to these static functions.

Comment (roconnor-blockstream) [reply C/jets.c:105]: Done.

File: C/jets.c [hunk 2]
```diff
@@ -80,6 +93,208 @@ bool sha_256_block(frameItem* dst, frameItem src, const txEnv* env) {
   return true;
 }
 
+/* Read bytes from a Simplicty bufffer of type (TWO^8)^<2^(n+1) into 'buf'.
```
Comment (apoelstra) [inline C/jets.c:96]: In c56cae90402f1fc04084957df690bf4f4784d94f:

Two typos: `Simplicty` -> `Simplicity` and `<2^(n+1)` I think think should be `(2^(n+1))`.

These are both repeated below.

Comment (apoelstra) [reply C/jets.c:96]: Oh, a third -- `bufffer` to `buffer` :).

**Review by apoelstra — APPROVED** [review]
ACK f9853eee38e0349d7949382c54155e6c17d44347 aside from comment nits.

I did not carefully check the type annotations this time, because they are enormous :). It also took me a lot of time to infer how this logic works and I am not 100% comfortable with it, but I think it's correct. At the very least it is cross-checked against Haskell, and sha2 is a very well-tested well-definied function.

File: C/jets.c (hunk 2 above)
Comment (roconnor-blockstream) [reply C/jets.c]: I've fixed the typos.  I've added some comments documenting the `X^<n` notation.

**Review by apoelstra — APPROVED** [review]
ACK 80980cbef9f3163f0616d62b8f293a5e1d0ba756

**Comment by apoelstra** [comment]
Thtanks, the new comments are super helpful!

**Pull request merged**